juntagrico/admins/job_admin.py

```diff
@@ -72,19 +72,10 @@ def copy_job_view(self, request, jobid):
     def get_queryset(self, request):
         return queryset_for_coordinator(self, request, 'type__activityarea__coordinator')
 
-    def formfield_for_foreignkey(self, db_field, request, **kwargs):
-        if db_field.name == 'type':
-            kwargs['queryset'] = JobTypeDao.visible_types()
-            kwargs = formfield_for_coordinator(request,
-                                               db_field.name,
-                                               'type',
-                                               'juntagrico.is_area_admin',
-                                               JobTypeDao.visible_types_by_coordinator,
-                                               **kwargs)
-            # show jobtype even if invisible to be able to edit and save this job with the same type
-            # HACK: get instance via url argument
-            instance_pk = request.resolver_match.kwargs.get('object_id')
-            if instance_pk is not None:
-                kwargs['queryset'] |= JobType.objects.filter(recuringjob__pk=instance_pk)
-                kwargs['queryset'] = kwargs['queryset'].distinct()
-        return super().formfield_for_foreignkey(db_field, request, **kwargs)
+    def get_type_queryset(self, request, obj):
+        visible_by_coordinator = formfield_for_coordinator(
+            request, None, None, 'juntagrico.is_area_admin',
+            JobTypeDao.visible_types_by_coordinator,
+            queryset=JobTypeDao.visible_types()
+        )['queryset']
+        return (JobType.objects.filter(recuringjob=obj) | visible_by_coordinator).distinct()
```

Comment:
why not dao?

Reply:
daos will be replaced by custom querysets in the future #527. So I avoid creating new ones. And this filter is simple enough as it is.